shub/utils.py

```diff
@@ -570,3 +573,87 @@ def download_from_pypi(dest, pkg=None, reqfile=None, extra_args=None):
                  target)
+
+
+@contextlib.contextmanager
+def update_yaml_dict(conf_path=None):
+    """
+    Context manager for updating a YAML file. Key ordering and comments are not
+    preserved.
+    """
+    if not conf_path:
+        click.secho("Using update_yaml_dict without path is deprecated. Import"
+                    " GLOBAL_SCRAPINGHUB_YML_PATH from shub.config",
+                    fg='yellow')
+        from shub.config import GLOBAL_SCRAPINGHUB_YML_PATH
+        conf_path = GLOBAL_SCRAPINGHUB_YML_PATH
+    try:
+        with open(conf_path, 'r') as f:
+            conf = yaml.safe_load(f) or {}
+    except IOError as e:
+        if e.errno != 2:
+            raise
+        conf = {}
+    # Code inside context manager is executed after this yield
+    yield conf
+    # Avoid writing "key: {}"
+    for key in list(conf.keys()):
+        if conf[key] == {}:
+            del conf[key]
+    with open(conf_path, 'w') as f:
+        # Avoid writing "{}"
+        if conf:
+            yaml.safe_dump(conf, f, default_flow_style=False)
+
+
+def has_project_access(project, endpoint, apikey):
+    """Check whether an API key has access to a given project. May raise
+    InvalidAuthException if the API key is invalid (but not if it is valid but
+    lacks access to the project)"""
+    conn = Connection(apikey, url=endpoint)
+    try:
+        return project in conn.project_ids()
+    except APIError as e:
+        if 'Authentication failed' in e.message:
+            raise InvalidAuthException
+        else:
+            raise RemoteErrorException(e.message)
+
+
+def create_scrapinghub_yml_wizard(conf, target='default', image=False):
+    """
+    Ask user for project ID, ensure they have access to that project, and save
+    it to given ``target`` in local ``scrapinghub.yml`` if desired.
+    """
+    closest_scrapycfg = closest_file('scrapy.cfg')
+    sh_yml_dir = (os.path.dirname(closest_scrapycfg) if closest_scrapycfg
+                  else os.getcwd())
+    closest_sh_yml = os.path.join(sh_yml_dir, 'scrapinghub.yml')
+    if target not in conf.projects:
+        # Get default endpoint and API key (meanwhile making sure the user is
+        # logged in)
+        endpoint, apikey = conf.get_endpoint(0), conf.get_apikey(0)
+        project = click.prompt("Target project ID", type=int)
+        if not has_project_access(project, endpoint, apikey):
+            raise InvalidAuthException(
+                "The account you logged in to has no access to project {}. "
+                "Please double-check the project ID and make sure you logged "
+                "in to the correct acount.".format(project),
+            )
+        conf.projects[target] = project
+    if image and target not in conf.images:
+        repository = click.prompt(
+            "Image repository (leave empty to use Scrapinghub's repository)",
+            default=True, show_default=False)
+        conf.images[target] = repository
```

Comment:
Remark: I'm fine with the changes but would like to clarify - according to the code wizard adds `(target, repo)` pair to `images` section, but I recall we agreed to move away from `images` section step-by-step, why not add it as `image: repo` under the specific project section?